skyportal/handlers/api/source.py

```diff
@@ -1221,6 +1234,56 @@ class Validator(Schema):
                     Obj.id == classification_subquery.c.obj_id,
                     isouter=True,
                 )
+        if nonclassifications is not None:
+            if isinstance(nonclassifications, str) and "," in nonclassifications:
+                nonclassifications = [c.strip() for c in nonclassifications.split(",")]
+            elif isinstance(nonclassifications, str):
+                nonclassifications = [nonclassifications]
+            else:
+                return self.error(
+                    "Invalid non-classifications value -- must provide at least one string value"
+                )
+            taxonomy_names, nonclassifications = list(
+                zip(
+                    *list(
+                        map(
+                            lambda c: (
+                                c.split(":")[0].strip(),
+                                c.split(":")[1].strip(),
+                            ),
+                            nonclassifications,
+                        )
+                    )
+                )
+            )
+            classification_accessible_query = (
+                Classification.query_records_accessible_by(self.current_user).subquery()
+            )
+
+            classification_query = (
+                DBSession()
```

Comment:
I would call the first one `classification_accessible_subquery` and the second one `unclassifications_subquery` to make the logic a bit clearer.

Reply:
yes good idea